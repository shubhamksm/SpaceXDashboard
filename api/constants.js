export const BASE_URL = 'https://api.spaceXdata.com/v3/launches?limit=1&launch_success=true&land_success=true&launch_year=2014';