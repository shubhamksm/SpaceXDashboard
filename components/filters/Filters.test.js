import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Filters from './';
import { filtersLaunchYears } from '../../constants';

describe("Snapshot Test - Filters", () => {
  it('Should show all filters', () => {
    const component = renderer.create(<Filters
      heading="Launch Year"
      _key="launch_year"
      filterData={filtersLaunchYears}
      selected={2006}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Content Check - Filters", () => {
  it('Should have correct details and classNames', () => {
    const component = shallow(<Filters
      heading="Launch Year"
      _key="launch_year"
      filterData={filtersLaunchYears}
      selected={2006}
    />);

    expect(component.find("header").text()).toEqual("Launch Year");
    expect(component.find(".selected").text()).toEqual("2006");
  });
});