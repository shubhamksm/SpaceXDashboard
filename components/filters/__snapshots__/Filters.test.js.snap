// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test - Filters Should show all filters 1`] = `
<article>
  <header>
    Launch Year
  </header>
  <hr />
  <div
    className="filterCard"
  >
    <button
      className="selected"
      onClick={[Function]}
    >
      2006
    </button>
    <button
      onClick={[Function]}
    >
      2007
    </button>
    <button
      onClick={[Function]}
    >
      2008
    </button>
    <button
      onClick={[Function]}
    >
      2009
    </button>
    <button
      onClick={[Function]}
    >
      2010
    </button>
    <button
      onClick={[Function]}
    >
      2011
    </button>
    <button
      onClick={[Function]}
    >
      2012
    </button>
    <button
      onClick={[Function]}
    >
      2013
    </button>
    <button
      onClick={[Function]}
    >
      2014
    </button>
    <button
      onClick={[Function]}
    >
      2015
    </button>
    <button
      onClick={[Function]}
    >
      2016
    </button>
    <button
      onClick={[Function]}
    >
      2017
    </button>
    <button
      onClick={[Function]}
    >
      2018
    </button>
    <button
      onClick={[Function]}
    >
      2019
    </button>
    <button
      onClick={[Function]}
    >
      2020
    </button>
  </div>
</article>
`;
